test/integration/master/audit_test.go

```diff
@@ -0,0 +1,343 @@
+			auditinternal.StageResponseComplete,
+			fmt.Sprintf("/api/v1/namespaces/%s/configmaps", namespace),
+			"create",
+			201,
+			auditTestUser,
+			"configmaps",
+			namespace,
+			true,
+			true,
+			"allow",
+		}, {
+			auditinternal.LevelRequestResponse,
+			auditinternal.StageResponseComplete,
+			fmt.Sprintf("/api/v1/namespaces/%s/configmaps/audit-configmap", namespace),
+			"get",
+			200,
+			auditTestUser,
+			"configmaps",
+			namespace,
+			false,
+			true,
+			"allow",
+		}, {
+			auditinternal.LevelRequestResponse,
+			auditinternal.StageResponseComplete,
+			fmt.Sprintf("/api/v1/namespaces/%s/configmaps", namespace),
+			"list",
+			200,
+			auditTestUser,
+			"configmaps",
+			namespace,
+			false,
+			true,
+			"allow",
+		}, {
+			auditinternal.LevelRequestResponse,
+			auditinternal.StageResponseStarted,
+			fmt.Sprintf("/api/v1/namespaces/%s/configmaps?timeoutSeconds=%d&watch=true", namespace, watchTestTimeout),
+			"watch",
+			200,
+			auditTestUser,
+			"configmaps",
+			namespace,
+			false,
+			false,
+			"allow",
+		}, {
+			auditinternal.LevelRequestResponse,
+			auditinternal.StageResponseComplete,
+			fmt.Sprintf("/api/v1/namespaces/%s/configmaps?timeoutSeconds=%d&watch=true", namespace, watchTestTimeout),
+			"watch",
+			200,
+			auditTestUser,
+			"configmaps",
+			namespace,
+			false,
+			false,
+			"allow",
+		}, {
+			auditinternal.LevelRequestResponse,
+			auditinternal.StageResponseComplete,
+			fmt.Sprintf("/api/v1/namespaces/%s/configmaps/audit-configmap", namespace),
+			"update",
+			200,
+			auditTestUser,
+			"configmaps",
+			namespace,
+			true,
+			true,
+			"allow",
+		}, {
+			auditinternal.LevelRequestResponse,
+			auditinternal.StageResponseComplete,
+			fmt.Sprintf("/api/v1/namespaces/%s/configmaps/audit-configmap", namespace),
+			"patch",
+			200,
+			auditTestUser,
+			"configmaps",
+			namespace,
+			true,
+			true,
+			"allow",
+		}, {
+			auditinternal.LevelRequestResponse,
+			auditinternal.StageResponseComplete,
+			fmt.Sprintf("/api/v1/namespaces/%s/configmaps/audit-configmap", namespace),
+			"delete",
+			200,
+			auditTestUser,
+			"configmaps",
+			namespace,
+			true,
+			true,
+			"allow",
+		},
+	}
+
+	ok, err := checkAuditLines(t, logFile.Name(), expectedEvents, version)
+	if err != nil {
+		t.Fatalf("Unexpected error: %v", err)
+	}
+	if !ok {
+		t.Fatalf("Failed to match all expected events")
+	}
+}
+
+type auditEvent struct {
+	level             auditinternal.Level
+	stage             auditinternal.Stage
+	requestURI        string
+	verb              string
+	code              int32
+	user              string
+	resource          string
+	namespace         string
+	requestObject     bool
+	responseObject    bool
+	authorizeDecision string
+}
+
+func expectNoError(t *testing.T, err error, msg string) {
+	if err != nil {
+		t.Fatalf("%s: %v", msg, err)
+	}
+}
+
+func checkAuditLines(t *testing.T, logFile string, expected []auditEvent, version string) (bool, error) {
```

Comment:
This looks copied from https://github.com/kubernetes/kubernetes/blob/cef2d325ee1be894e883d63013f75cfac5cb1246/test/e2e/auth/audit.go#L745

Can we just extract it to [test/utils/audit.go](https://github.com/kubernetes/kubernetes/tree/master/test/utils) ?
(and change this function to take an `io.Reader` instead of the framework / logFile)
(and maybe change the "allFound" bool return to a "missingEvents" return)

Reply:
Done.